File: september27_closest_pair_two_arrays.py
#User function Template for python3
import sys
class Solution:
    def printClosest (self,arr, brr, n, m, x) : 
        #code here
        i,j,ans=0,m-1,sys.maxsize
        ls=[0,0]
        while i<n and j>=0:
            sum=arr[i]+brr[j]
            res=abs(sum-x)
            if(ans>res):
                ans=res
                ls[0]=arr[i]
                ls[1]=brr[j]
            elif sum<x:
                i+=1
            else:
                j-=1
                    
        return ls


#{ 
 # Driver Code Starts
#Initial Template for Python 3

for _ in range(0,int(input())):
    n, m = map(int, input().split())
    arr = list(map(int, input().strip().split()))
    brr = list(map(int, input().strip().split()))
    x = int(input())
    ob = Solution()
    ans = ob.printClosest(arr, brr, n, m, x)
    print(abs(ans[0]+ans[1] - x))
# } Driver Code Ends

File: september27_closest_pair_two_arrays.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function teamplate for C++

class Solution
{
public:
    vector<int> printClosest(int arr[], int brr[], int n, int m, int x)
    {
        // code here
        vector<int> res(2, 0);
        int i = 0, j = m - 1, ans = INT_MAX;

        while (i < n && j >= 0)
        {
            int sum = arr[i] + brr[j];
            if (ans > (abs(sum - x)))
            {
                ans = abs(sum - x);
                res[0] = arr[i];
                res[1] = brr[j];
            }
            if (sum > x)
                j--;
            else
                i++;
        }
        return res;
    }
};

//{ Driver Code Starts.

int main()
{
    int t;
    cin >> t;
    int m, n, x;
    while (t--)
    {
        cin >> n >> m;
        int a[n], b[m];

        for (int i = 0; i < n; i++)
            cin >> a[i];

        for (int i = 0; i < m; i++)
            cin >> b[i];

        cin >> x;

        vector<int> ans;
        Solution ob;
        ans = ob.printClosest(a, b, n, m, x);
        cout << abs(ans[0] + ans[1] - x) << endl;
    }
    return 0;
}

// } Driver Code Ends